### ISSUE TITLE
Add http_endpoint_configuration to r/aws_kinesis_firehose_delivery_stream

### ISSUE THREAD

**Issue by radamisa** [hashicorp/terraform-provider-aws#14469]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Amazon Kinesis Data Firehose HTTP endpoint delivery was introduced to Firehose a week ago. This is a request to add `http_endpoint_configuration` to `aws_kinesis_firehose_delivery_stream`.

### New or Affected Resource(s)

r/aws_kinesis_firehose_delivery_stream

### Potential Terraform Configuration

```hcl
resource "aws_kinesis_firehose_delivery_stream" "test_stream" {
  name        = "terraform-kinesis-firehose-test-stream"
  destination = "http_endpoint"

  s3_configuration { // Similar to implementation in splunk_configuration, this maps to HttpEndpointDestinationConfiguration.S3Configuration
    role_arn           = aws_iam_role.firehose.arn
    bucket_arn         = aws_s3_bucket.bucket.arn
    buffer_size        = 10
    buffer_interval    = 400
    compression_format = "GZIP"
  }

  http_endpoint_configuration {
    url        = "https://api.example.com:443"
    name       = "MyHTTPEndpoint"
    access_key = "my-key"
    buffering_hints {
      size_in_mb = 5
      interval_in_seconds = 300
    }
   // cloudwatch_logging_options is CloudWatchLoggingOptions as used in s3_configuration
   // processing_configuration is ProcessingConfiguration as used in extended_s3_configuration
   role_arn       = "arn:my_role"
   s3_backup_mode = "FailedDataOnly"
  }
}
```

### References

* https://docs.aws.amazon.com/firehose/latest/APIReference/API_CreateDeliveryStream.html#Firehose-CreateDeliveryStream-request-HttpEndpointDestinationConfiguration
* https://aws.amazon.com/blogs/big-data/analyze-logs-with-datadog-using-amazon-kinesis-data-firehose-http-endpoint-delivery/

**Comment by ewbankkit** [comment]
[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/07/amazon-kinesis-data-firehose-now-supports-data-delivery-to-http-endpoints/).

Requires [AWS SDK v1.33.15](https://github.com/aws/aws-sdk-go/releases/tag/v1.33.15):

* #14000

**Comment by shuheiktgw** [comment]
I've finally started to work on this issue...

**Comment by ZsoltPath** [comment]
I'd work on it for a few days if nobody else. 
I hope I can open a PR then.

**Comment by ZsoltPath** [comment]
Please vote on the PR linked above to sort his out!

**Comment by bflad** [comment]
Initial support for this functionality has been merged and will release with version 3.15.0 of the Terraform AWS Provider, likely later next week. Thanks to @ZsoltPath for the implementation. 👍
